src/Dynode/model/core/socket/inputSocket.ts

```diff
@@ -128,30 +123,25 @@ class InputSocket extends Socket {
   /**
    * Check if object has a set value.
    */
-  isSet(): boolean {
+  public isSet(): boolean {
     if (this.linkedSocket) {
       return super.isSet();
     }
-    return this.hasDefault;
+    return this.isDefaultSet();
```

Comment:
If we treat unlinked input sockets as nothing, then unlinked input sockets are always set. So, just return true here.